### PR TITLE
close #518: remove Typing.tla, AssumeType, and ##

### LINKED ISSUE #518: [Refactoring] Remove Typing!AssumeType, Typing!##
Java-like annotations are a much more convenient tool for writing type annotations. When using annotations we can simplify the intermediate language of the type checker. To keep things simple, we should remove `AssumeType` and `##`.

### PR DESCRIPTION
**Pull request by konnov**
This is more like an anti-feature PR, as it removes the code that processes the first version of the annotations that we developed for the new type checker. As java-like annotations look much better, most likely, nobody will even notice.

- [ ] ~Tests added for any new code~
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] ~Documentation added for any new functionality~
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -10,6 +10,11 @@
          * Some bug fix, see #124
 
      DO NOT LEAVE A BLANK LINE BELOW THIS PREAMBLE -->
+
+### Features
+
+* Type checker: removed `Typing.tla`, `AssumeType`, and `##`, see #518
```
Comment (shonfeder) [inline UNRELEASED.md:16]: Maybe this should go in a `Removed` section?

Just thinking out loud, but we might start using the "keep a changelog" suggested sections: https://keepachangelog.com/en/1.0.0/#how

**Review by shonfeder — APPROVED** [review]

File: UNRELEASED.md (hunk 1 above)
Comment (konnov) [reply UNRELEASED.md:16]: Yep, "Removed" sounds good

Comment (konnov) [reply UNRELEASED.md]: > Just thinking out loud, but we might start using the "keep a changelog" suggested sections:

Comment (konnov) [reply UNRELEASED.md]: That is a good idea

**Pull request merged**